tools/src/lib.rs

```diff
@@ -13,6 +15,38 @@ pub struct Address {
     pub postcode: Option<String>,
 }
 
+impl Address {
+    pub const NB_FIELDS: usize = 9;
+
+    pub fn count_non_empty_fields(&self) -> usize {
+        2 + self.number.is_some() as usize
```

Comment:
Why this 2?

Reply:
It corresponds to longitude and latitude, even they are not optional it gives a result closer to what we expect this function to do. I just added a tiny comment to make it clearer.

Reply:
Great!